File: AlterAmbient/FormAnadirProyecto.cs
using IAHRIS.BBDD;
using Microsoft.VisualBasic;
using Microsoft.VisualBasic.CompilerServices;
using System;
using System.Data;
using System.Windows.Forms;

namespace IAHRIS
{
    public partial class FormAnadirProyecto
    {
        int _idProyecto;
        private BBDD.OleDbDataBase _cMDB;
        private MultiLangXML.MultiIdiomasXML _traductor;

        public FormAnadirProyecto()
        {
            InitializeComponent();
            _btnAceptar.Name = "btnAceptar";
        }

        public FormAnadirProyecto(OleDbDataBase MDB, bool editar, int p=0)
        {
            Form argform = this;

            // This call is required by the Windows Form Designer.
            InitializeComponent();

            // Add any initialization after the InitializeComponent() call.
            _cMDB = MDB;

            _btnEditar.Name = "btnEditar";
            _btnAceptar.Name = "btnAceptar";

            // -------------------------------------
            // ---- Traducir formulario ------------
            // -------------------------------------
            _traductor = MultiLangXML.MultiIdiomasXML.Instancia;
            _traductor.TraducirForm(this);

            _btnEditar.Enabled = false;

            if (editar == true) 
            {
                _btnEditar.Enabled = true;
                _btnAceptar.Enabled = false;
                _idProyecto = p;
                
                //obtener proyecto seleccionado y rellenar campos                
                string str = "SELECT * FROM [Proyecto] WHERE ID_Proyecto = " + p;
                var ds = _cMDB.RellenarDataSet("Proyectos", str);
                var dr = ds.Tables[0].Rows[0];

                txtNombre.Text = dr[1].ToString();
                txtDescripcion.Text = dr[2].ToString();      
            }
        }       


        private void btnAceptar_Click(object sender, EventArgs e)
        {
            var valores = new[] { txtNombre.Text, txtDescripcion.Text };
            var campos = new[] { "nombre", "descripcion" };

            //Validaciones
            if (string.IsNullOrEmpty(txtNombre.Text) & !string.IsNullOrEmpty(txtDescripcion.Text))            
                MessageBox.Show(_traductor.traducirMensaje(MultiLangXML.MultiIdiomasXML.TIPO_MENSAJE.M_ERROR, "strErrorValidationProj"), "Error", MessageBoxButtons.OK, MessageBoxIcon.Information);
          
            if (txtNombre.Text.Length > 20)           
                MessageBox.Show(_traductor.traducirMensaje(MultiLangXML.MultiIdiomasXML.TIPO_MENSAJE.M_ERROR, "strTooLongName"), _traductor.traducirMensaje(MultiLangXML.MultiIdiomasXML.TIPO_MENSAJE.M_ERROR, "strError"), MessageBoxButtons.OK, MessageBoxIcon.Error);




            //Insertar

            if (!ExisteProyecto(valores[0]))
            {
                if (!_cMDB.InsertarRegistro("Proyecto", campos, valores))        
                MessageBox.Show(_traductor.traducirMensaje(MultiLangXML.MultiIdiomasXML.TIPO_MENSAJE.M_ERROR,
                                    "strErrorInsertProj"), _traductor.traducirMensaje(MultiLangXML.MultiIdiomasXML.TIPO_MENSAJE.M_ERROR, 
                                    "strError"), MessageBoxButtons.OK, MessageBoxIcon.Error);
            }
            else
            {
                MessageBox.Show(_traductor.traducirMensaje(MultiLangXML.MultiIdiomasXML.TIPO_MENSAJE.M_ERROR, "strErrorProjectAlreadyExist"), _traductor.traducirMensaje(MultiLangXML.MultiIdiomasXML.TIPO_MENSAJE.M_ERROR, "strError"), MessageBoxButtons.OK, MessageBoxIcon.Error);

            }


            Dispose();
            return;
                     
        }

        private void btnEditar_Click(object sender, EventArgs e)
        {           
            string str = "UPDATE [Proyecto] " +
                         "SET nombre = '" + txtNombre.Text + "', descripcion = '" + txtDescripcion.Text +  "' " +
                         "WHERE ID_Proyecto = "+ _idProyecto;

            if (_cMDB.EjecutarSQL(str) == -1)                    
                MessageBox.Show(_traductor.traducirMensaje(MultiLangXML.MultiIdiomasXML.TIPO_MENSAJE.M_ERROR, "strErrorUpdateProj"),
                                _traductor.traducirMensaje(MultiLangXML.MultiIdiomasXML.TIPO_MENSAJE.M_ERROR, "strError"),
                                MessageBoxButtons.OK, MessageBoxIcon.Error);

            Dispose();
            return;
            
        }

        private bool ExisteProyecto(string nombreProyecto)
        {
            DataSet ds = null;
            DataRow dr;

            ds = _cMDB.RellenarDataSet("existe", "SELECT COUNT(*) FROM Proyecto WHERE nombre='" + nombreProyecto + "'");

            dr = ds.Tables[0].Rows[0];
            if (Conversions.ToBoolean(Operators.ConditionalCompareObjectGreater(dr[0], 0, false)))
            {
                return true;
            }            
            else
            {
                return false;
            }
        }
    }
}

File: AlterAmbient/FormAnadirProyecto.vb
Public Class FormAnadirProyecto

    Dim _cMDB As BBDD.OleDbDataBase
    Private _traductor As MultiLangXML.MultiIdiomasXML

    Public Sub New(ByVal MDB As BBDD.OleDbDataBase)
        ' This call is required by the Windows Form Designer.
        InitializeComponent()

        ' Add any initialization after the InitializeComponent() call.
        Me._cMDB = MDB

        ' -------------------------------------
        ' ---- Traducir formulario ------------
        ' -------------------------------------
        Me._traductor = New MultiLangXML.MultiIdiomasXML(Me)
        Me._traductor.traducirFormPorConf(Application.StartupPath, "\conf.xml")
    End Sub

    Private Sub btnAceptar_Click(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles btnAceptar.Click
        If (Me.txtNombre.Text <> "" And Me.txtDescripcion.Text <> "") Then
            If (Me.txtNombre.Text.Length > 20) Then
                MessageBox.Show(Me._traductor.traducirMensaje(MultiLangXML.MultiIdiomasXML.TIPO_MENSAJE.M_ERROR, "strTooLongName"), _
                                Me._traductor.traducirMensaje(MultiLangXML.MultiIdiomasXML.TIPO_MENSAJE.M_ERROR, "strError"), _
                                MessageBoxButtons.OK, MessageBoxIcon.Error)
                'MessageBox.Show("El campo NOMBRE no puede ser mayor de 20 caracteres", "ERROR", MessageBoxButtons.OK, MessageBoxIcon.Information)
                Return
            End If

            Dim valores As String() = {Me.txtNombre.Text, Me.txtDescripcion.Text}
            Dim campos As String() = {"nombre", "descripcion"}
            If (Me._cMDB.InsertarRegistro("Proyecto", campos, valores)) Then
                Me.Dispose()
            Else
                MessageBox.Show(Me._traductor.traducirMensaje(MultiLangXML.MultiIdiomasXML.TIPO_MENSAJE.M_ERROR, "strErrorInsertProj"), _
                                Me._traductor.traducirMensaje(MultiLangXML.MultiIdiomasXML.TIPO_MENSAJE.M_ERROR, "strError"), _
                                MessageBoxButtons.OK, MessageBoxIcon.Error)
                Return
            End If

        End If
    End Sub
End Class